ADJ-Internship/BusinessService/Implementations/PurchaseOrderService.cs

```diff
@@ -17,10 +18,233 @@ namespace ADJ.BusinessService.Implementations
+            }
+
+            await UnitOfWork.SaveChangesAsync();
+
+            var rs = Mapper.Map<OrderDetailDTO>(entity);
+            return rs;
+        }
+
+        public async void DeleteOrderDetailAsync(PagedListResult<OrderDetailDTO> orderDetailsView, int orderId)
+        {
+            var orderDetailResult = await _orderDetailDataProvider.ListAsync();
+            List<OrderDetail> orderDetails = orderDetailResult.Items;
+            
+
+            bool delete;
+            foreach (var item1 in orderDetails)
+            {
+                if (item1.OrderId == orderId)
+                {
+                    delete = true;
+                    foreach (var item2 in orderDetailsView.Items)
+                    {
+                        if ((item2.Id != 0) && (item1.Id == item2.Id))
+                        {
+                            delete = false;
+                        }
+                    }
+
+                    if (delete)
+                    {
+                        _orderDetailRepository.Delete(item1);
+                    }
+                }
+            }
+        }
+
+
+        public async Task<bool> UniquePONumAsync(string PONumber, int? id)
+        {
+            var orderResult = await _orderDataProvider.ListAsync();
+            List<Order> orders = orderResult.Items;
+
+            if (id == null || id == 0)
+            {
+                foreach (var item in orders)
+                {
+                    if (item.PONumber == PONumber)
+                    {
+                        return false;
+                    }
+                }
+            }
+            else
+            {
+                foreach (var item in orders)
+                {
+                    if ((item.PONumber == PONumber) && (item.Id != id))
+                    {
+                        return false;
+                    }
+                }
+            }
+       
+            return true;
+        }
+
+        public async Task<bool> UniqueItemNumAsync(string itemNum, int? id)
+        {
+            var orderDetailResult = await _orderDetailDataProvider.ListAsync();
+            List<OrderDetail> orderDetails = orderDetailResult.Items;
+
+            if (id == null || id == 0)
+            {
+                foreach (var item in orderDetails)
+                {
+                    if (item.ItemNumber == itemNum)
+                    {
+                        return false;
+                    }
+                }
+            }
+            else
+            {
+                foreach (var item in orderDetails)
+                {
+                    if ((item.ItemNumber == itemNum) && (item.Id != id))
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        public async Task<OrderDTO> GetOrderByPONumber(string poNumber)
+        {
+            var orderResult = await _orderDataProvider.ListAsync();
```

Comment:
don't use DataProvider for getting data for processing as it will load the whole tables
just use it for listing data (with filtering)
if you need to get something, create a method in Repository and use DbContext to get the data you need